### DISCUSSION TITLE
Help with [eme_bookings_report_link]

### DISCUSSION THREAD

**Discussion by roysinn** [liedekef/events-made-easy#286] [Q&A] [answered]
Created a page with the following URL
https://lcar.ca/bookings/ 
The Page code is:
[eme_events template_id=12]

**Template 12 code is:**
```
<h5>#_EVENTNAME</h5>
#_STARTDATE{l, F j, Y}
#_STARTDATE{g:i A T} (Pacific/Vancouver Time)
Total Seats: #_TOTALSEATS
Registered: #_RESERVEDSEATS
Available Seats: #_AVAILABLESEATS
<br/><strong>Bookings</strong>
#_BOOKINGS 
[eme_bookings_report_link title="Download CSV of bookings" event_id=#EVENT_ID template_id=13]
```

**And, Template 13 code is:**
`#_LASTNAME,#_FIRSTNAME,#_SEATS,#_EMAIL,#_PHONE`

Everything before the "[eme_bookings_report_link titl....." line works and the "[eme_bookings_report_link titl....." line generates a link.  But clicking the link simply returns the original page with no CSV download. It does, however, change the URL from this:
`https://lcar.ca/bookings/ `
 to this
`https://lcar.ca/bookings/?event_id=#EVENT_ID&template_id=13&template_id_header=0&public_access&eme_bookings=report`

I was expecting/hoping to get a download prompt for a CSV file. 

Something wrong with my code or expectations? 

Thanks for any help.

**Comment by liedekef** [comment, marked as answer]
#EVENT_ID does't exist as placeholder, it is #_EVENTID

**Comment by roysinn** [comment]
That was it. Works like a charm now. Thanks Franky.

**Comment by roysinn** [comment]
Now having trouble with the format of the data in the downloaded CSV. Here's what I've specified:
Header Template: #_LASTNAME,#_FIRSTNAME,#_SEATS,#_EMAIL,#_PHONE
Template: '#_LASTNAME','#_FIRSTNAME','#_SEATS,#_EMAIL','#_PHONE'
Header and content of one line in the CSV:
#_LASTNAME;"#_FIRSTNAME";"#_SEATS";"#_EMAIL";"#_PHONE"
'Smith';"'Jack'";"'1";"jsmith@mail.com'";"'8883211234'"

Looks like it is using ";" as the delimiter even though "," is specified as the "CSV separator" in Settings/Other. 

Notice anything amiss in my templates? 

Thanks again in advance!
Roy

**Reply by liedekef** [reply]
This is not related to the original question, please open a new thread in the future!
Concerning the question, the generated csv indeed always used ";" there, this change fixes it:
https://github.com/liedekef/events-made-easy/commit/ed8faaea4a4daf33a6dbd533eb1a0df3d8bf9e0e

Next to that: your template should be without single quotes, and your header template should contain text, not placeholders

**Reply by roysinn** [reply]
Many thanks again Franky. Sorry for not starting a new thread.
